application/clicommands/ScanCommand.php

```diff
@@ -58,9 +70,44 @@ public function indexAction()
                 $name
             );
 
+            $sinceLastScan = $this->processThresholdToDateTiime($this->params->get('since-last-scan'));
+
+            if ($sinceLastScan !== null) {
+                $this->cleanupLastScanTargets($this->getDb(), $sinceLastScan);
```

Comment:
I would just call the method `cleanupTargets()` without any params and let the method decide whether to remove targets or not.